common/lib/common-utils/src/fluidError.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IFluidError,
+    ITelemetryProperties,
+    ISensitiveDebugData,
+    isIFluidError,
+} from "@fluidframework/common-definitions";
+
+export abstract class FluidError extends Error implements IFluidError {
```

Comment:
Can we make this change more incremental, and not add layers, but change existing layers? There is already LoggingError that does same thing, can you please change it instead of adding new class?

As mentioned below, we would need to find all places where we patch error object (like in createOdspNetworkError) and patch them by adding method addProperty() to LoggingError, change interfaces IDriverErrorBase, IOdspError, IErrorBase to expose addProperty() method and change all places where we are patching error object).

Also all throttlingError -> driver.throttlingError changes should go into different PR.

The more incremental you move, the more chances we will not break anything :)

Reply:
This will be very incremental.  This PR is just to get feedback on the direction - so, thank you!  Once there's consensus I'll open a handful of issues and hopefully hand some of them out to others on the team who can help.  First change is to add UTs to cover all the spots we've discussed so we know precisely what behavior is changing as we refactor.